pkg/controller/migmigration/task.go

```diff
@@ -386,7 +409,39 @@ func (t *Task) Run() error {
 		if err = t.next(); err != nil {
 			return liberr.Wrap(err)
 		}
+	case CreateDirectImageMigration:
+		// Create the DirectImageMigration CR
+		err := t.createDirectImageMigration()
+		if err != nil {
+			return liberr.Wrap(err)
+		}
+		if err = t.next(); err != nil {
+			return liberr.Wrap(err)
+		}
+	case WaitForDirectImageMigrationToComplete:
+		// Get the DirectImageMigration CR
+		dim, err := t.getDirectImageMigration()
+		if err != nil {
+			return liberr.Wrap(err)
+		}
+		if dim == nil {
+			return errors.New("DirectImageMigration not found")
+		}
 
+		completed, reasons := dim.HasCompleted()
+		t.Log.Info("is migrations", "name", dim.Name, "completed", completed, "phase", dim.Status.Phase)
+
+		if completed {
+			if len(reasons) > 0 {
+				t.fail(MigrationFailed, reasons)
```

Comment:
So the whole migration is considered to be failed if any image migrations fail, is this correct?

Reply:
That is the way we originally wrote this. It's possible we want to revisit given the recent move to parallel execution. Originally this ran and finished first, so the thinking was a failure here would be following the "fail early" model. @dymurray should we modify this to follow the "Completed with warnings" approach? I guess the same question would apply to dvm.

Reply:
+1 I think this should follow the "Completed with warnings" approach.... though there is a concern that all the resources that get restored which depend upon the volume/image may be borked.

But I suppose that's consistent with our other approach returning "Completed with warnings" so I think thats fine

Reply:
I think we're in agreement that we should move to the "Completed With Warnings" behavior, but lets handle that as follow-on issues. I've filed separate issues for DIM and DVM:
https://github.com/konveyor/mig-controller/issues/790
https://github.com/konveyor/mig-controller/issues/789